doc/reference/configuration/configuration_reference.rst

```diff
@@ -3,13 +3,280 @@
+
+
+.. _configuration_reference_replicasets_name_config_parameter:
+
+.. confval:: replicasets.<replicaset_name>.<config_parameter>
+
+    Any configuration parameter that can be defined in the replica set :ref:`scope <configuration_scopes>`.
+    For example, :ref:`iproto <configuration_reference_iproto>` and :ref:`database <configuration_reference_database>` configuration parameters defined at the replica set level are applied to all instances in this replica set.
+
+
+
+..  _configuration_reference_instances:
+
+instances
+*********
+
+.. NOTE::
+
+    ``instances`` can be defined in the replica set :ref:`scope <configuration_scopes>` only.
+
+-   :ref:`instances.\<instance_name\> <configuration_reference_instances_name>`
+-   :ref:`instances.\<instance_name\>.\<config_parameter\> <configuration_reference_instances_name_config_parameter>`
+
+.. _configuration_reference_instances_name:
+
+.. confval:: instances.<instance_name>
+
+    An instance name.
+
+
+.. _configuration_reference_instances_name_config_parameter:
+
+.. confval:: instances.<instance_name>.<config_parameter>
+
+    Any configuration parameter that can be defined in the instance :ref:`scope <configuration_scopes>`.
+    For example, :ref:`iproto <configuration_reference_iproto>` and :ref:`database <configuration_reference_database>` configuration parameters defined at the instance level are applied to this instance only.
+
+
+..  _configuration_reference_credentials:
+
+credentials
+-----------
+
+..  TODO: https://github.com/tarantool/doc/issues/3666
+
+.. NOTE::
+
+    ``credentials`` can be defined in any :ref:`scope <configuration_scopes>`.
+
+
+..  _configuration_reference_credentials_role:
+
+roles
+~~~~~
+
+.. _configuration_reference_credentials_roles:
+
+.. confval:: credentials.roles
+
+    | Type: map
+    | Default: nil
+    | Environment variable: TT_CREDENTIALS_ROLES
+
+.. _configuration_reference_credentials_roles_name_roles:
+
+.. confval:: credentials.roles.<role_name>.roles
+
+
+.. _configuration_reference_credentials_roles_name_privileges:
+
+.. confval:: credentials.roles.<role_name>.privileges
+
+    See :ref:`privileges <configuration_reference_credentials_privileges>`.
+
+
+..  _configuration_reference_credentials_user:
+
+users
+~~~~~
+
+
+.. _configuration_reference_credentials_users:
+
+.. confval:: credentials.users
+
+    | Type: map
+    | Default: nil
+    | Environment variable: TT_CREDENTIALS_USERS
+
+
+.. _configuration_reference_credentials_users_name_password:
+
+.. confval:: credentials.users.<username>.password
+
+
+.. _configuration_reference_credentials_users_name_roles:
+
+.. confval:: credentials.users.<username>.roles
+
+
+.. _configuration_reference_credentials_users_name_privileges:
+
+.. confval:: credentials.users.<username>.privileges
+
+    See :ref:`privileges <configuration_reference_credentials_privileges>`.
+
+
+..  _configuration_reference_credentials_privileges:
+
+privileges
+~~~~~~~~~~
```

Comment:
It is a kind of template with two references to it (from `credentials.users.<username>.privileges` and `credentials.roles.<rolename>.privileges`). However, visually it looks like an ordinary sub-section within the hierarchy.

I think we should mark such templates somehow: some symbols like `<privileges>` (that looks a bit ugly for me, but anyway better) or some block with borders around.